Lib/platform.py

```diff
@@ -797,6 +802,16 @@ def __iter__(self):
             (self.processor,)
         )
 
+    @classmethod
+    def _make(cls, iterable):
+        # override factory to affect length check
+        num_fields = len(cls._fields)
+        result = cls.__new__(cls, *iterable)
+        if len(result) != num_fields + 1:
+            msg = f'Expected {num_fields} arguments, got {len(result)}'
+            raise TypeError(msg)
+        return result
+
     def __getitem__(self, key):
         return tuple(iter(self))[key]
```

Comment:
`iter()` is not needed.

And why not return just 6 in `__len__()`?

Reply:
> And why not return just 6 in `__len__()`?

I'm trying to avoid hard-coded static constants and implicit dependencies.

Reply:
Fixed in 55d310e6fa566f2baf09ede380d3d758a5305d82.